### PR TITLE
[SPARK-24341][FOLLOWUP][DOCS] Add migration note for IN subqueries behavior

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

The PR updates the migration guide in order to explain the changes introduced in the behavior of the IN operator with subqueries, in particular, the improved handling of struct attributes in these situations.

## How was this patch tested?

NA

### CODE REVIEW

**Comment by mgaido91** [comment]
cc @cloud-fan

File: docs/sql-programming-guide.md [hunk 1]
```diff
@@ -1879,6 +1879,7 @@ working with timestamps in `pandas_udf`s to get the best performance, see
 
 ## Upgrading From Spark SQL 2.3 to 2.4
 
+  - Since Spark 2.4, when there is a struct field in front of the IN operator, the inner query must contain a struct field as well. In previous versions, instead, the fields of the struct were compared to the output of the inner query. Eg. if `a` is a `struct(a string, b int)`, in Spark 2.4 `a in (select (1 as a, 'a' as b) from range(1))` is a valid query, while `a in (select 1, 'a' from range(1))` is not. In previous version it was the opposite.
```
Comment (cloud-fan) [inline docs/sql-programming-guide.md]: `IN operator` => `IN operator before a subquery`

**Comment by cloud-fan** [comment]
LGTM, thanks for adding it!

**Comment by mgaido91** [comment]
thank you for pointing this out @cloud-fan

**Comment by cloud-fan** [comment]
thanks, merging to master/2.4!

**Pull request closed without merging**